### DISCUSSION TITLE
Using zoxide content in copy operations

### DISCUSSION THREAD

**Discussion by maru-sama** [ajeetdsouza/zoxide#724] [Q&A] [answered]
Hi,

I wonder if it is somehow possible to use zoxide completion for other tasks than changing the directory. One example would be copying files to a project folder.

```
cp asset.jpg projectname<keybinding>
```

would then complete "projectname" to the project folder in question.
I guess this is more of a shell completion question though, but I thought I would ask here.

**Comment by jqhr** [comment]
use sub command.
Just like
```fish
cp asset.jpg (zoxide query i project)
```
For fish shell

**Comment by nezmaka** [comment]
If you use zsh, put the following in `.zshrc`:

```
function zcp() {
 cp "$1" $(zoxide query "$2")
}

function zmv() {
 mv "$1" $(zoxide query "$2")
}
```

This would allow you to e.g. run `zcp asset.jpg myproject` to copy a file.

**Reply by nezmaka** [reply]
These two would echo what the zxide query resolve to and prompt for a confirmation before execution (or aborts if it did not resolve to anything):

``` zsh
function zcp() {
    local src="$1"
    local dest

    # Query zoxide and check if it succeeds
    if ! dest=$(zoxide query "$2" 2>/dev/null); then
        echo "Error: No zoxide match found for '$2'."
        return 1
    fi

    echo "Destination resolved to: $dest"
    read -q "confirm?Proceed with copying? (y/n): "
    echo
    if [[ "$confirm" == "y" ]]; then
        cp "$src" "$dest"
        echo "Copied $src to $dest"
    else
        echo "Aborted"
    fi
}


function zmv() {
    local src="$1"
    local dest

    # Query zoxide and check if it succeeds
    if ! dest=$(zoxide query "$2" 2>/dev/null); then
        echo "Error: No zoxide match found for '$2'."
        return 1
    fi

    echo "Destination resolved to: $dest"
    read -q "confirm?Proceed with moving? (y/n): "
    echo
    if [[ "$confirm" == "y" ]]; then
        mv "$src" "$dest"
        echo "Moved $src to $dest"
    else
        echo "Aborted"
    fi
}
```

**Comment by ajeetdsouza** [comment, marked as answer]
I would not recommend using zoxide like this - queries are fuzzy by definition, and there's a chance that you will end up misplacing or overwriting a file.

**Reply by nezmaka** [reply]
That's true. I always first verify that the query goes where I want. Jumping back and executing the copy is still faster than any alternatives I'm currently aware of.
